qcodes/instrument_drivers/devices.py

```diff
@@ -1,6 +1,190 @@
+    value, and store the gain/division in the metadata.
+
+    Examples:
+        Resistive voltage divider
+        >>> vd = ParameterScaler(dac.chan0, division = 10)
+
+        Voltage multiplier
+        >>> vb = ParameterScaler(dac.chan0, gain = 30, name = 'Vb')
+
+        Transimpedance amplifier
+        >>> Id = ParameterScaler(multimeter.amplitude, division = 1e6, name = 'Id', unit = 'A')
+
+    Args:
+        output: Physical Parameter that need conversion
+        division: the division value
+        gain: the gain value
+        label: label of this parameter, by default uses 'output' label
+            but attaches _amplified or _attenuated depending if gain
+            or division has been specified
+        name: name of this parameter, by default uses 'output' name
+            but attaches _amplified or _attenuated depending if gain
+            or division has been specified
+        unit: resulting unit. It uses the one of 'output' by default
+    """
+
+    def __init__(self,
+                 output: Parameter,
+                 division: Union[int, float, Parameter] = None,
+                 gain: Union[int, float, Parameter] = None,
+                 name: str=None,
+                 label: str=None,
+                 unit: str=None) -> None:
+        self._wrapper_param = output
+
+        # Set the role, either as divider or amplifier
+        # Raise an error if nothing is specified
+        is_divider = division is not None
+        is_amplifier = gain is not None
+
+        if not (is_divider ^ is_amplifier):
+            raise ValueError('Provide only division OR gain')
+
+        if is_divider:
+            self.role = Role.DIVISION
+            self._multiplier = division
+        elif is_amplifier:
+            self.role = Role.GAIN
+            self._multiplier = gain
+
+        # Set the name
+        if name:
+            self.name = name
+        else:
+            if self.role == Role.DIVISION:
+                self.name = "{}_attenuated".format(self._wrapper_param.name)
+            elif self.role == Role.GAIN:
+                self.name = "{}_amplified".format(self._wrapper_param.name)
+
+        # Set label
+        if label:
+            self.label = label
+        elif name:
+            self.label = name
+        else:
+            if self.role == Role.DIVISION:
+                self.label = "{}_attenuated".format(self._wrapper_param.label)
+            elif self.role == Role.GAIN:
+                self.label = "{}_amplified".format(self._wrapper_param.label)
+
+        # Set the unit
+        if unit:
+            self.unit = unit
+        else:
+            self.unit = self._wrapper_param.unit
+
+        super().__init__(
+            name=self.name,
+            instrument=getattr(self._wrapper_param, "_instrument", None),
+            label=self.label,
+            unit=self.unit,
+            metadata=self._wrapper_param.metadata)
```

Comment:
What would be the result of implementing metadata inclusion like this? Are the attributes of the ‘_wrapper_parameter’ going to be included into the snapshot *next to* the attributes of the scaler, or in a separate field? I prefer the latter case.

Reply:
I removed the metadata inclusion from the parent. As long as the parent instrument is noted in the metadata, there is no need to copy its metadata